### PR TITLE
Add send-env and set-env options to ssh command

### PR DESCRIPTION
**Pull request by aacebedo**
Having similar options as SetEnv and SendEnv from the ssh client can be useful.

This PR adds those options to the ssh command.

### CODE REVIEW

**Comment by pascalbreuninger** [comment]
Hey @aacebedo, thanks for the PR!
I'm wondering how this differs from the `--workspace-env` flag in `devpod up`, except that you can set env vars per-connection.
 
Do you have a specific use case in mind?

**Comment by aacebedo** [comment]
My UC is the following.
I start a container during a CI step through a common template and I don't know before being actually used in the step which variables will be used.

With this, I am able to inject directly the variable I need when I write it.
In addition having temporary variables valid for a single command seems better than having to create a new container for each command (when using ssh --command option)

**Comment by bkneis** [comment]
LGTM, both options are supported by SSH so I don't see why we shouldn't support

**Review by bkneis — APPROVED** [review]

File: cmd/ssh.go [hunk 1]
```diff
@@ -38,6 +38,8 @@ type SSHCmd struct {
 	ForwardPortsTimeout string
 	ForwardPorts        []string
 	ReverseForwardPorts []string
+	SentEnvVars         []string
```
Comment (pascalbreuninger) [inline cmd/ssh.go:41]: small nit: 
```suggestion
	SendEnvVars         []string
```

File: pkg/ssh/helper.go [hunk 2]
```diff
@@ -90,13 +90,22 @@ func ConfigFromKeyBytes(keyBytes []byte) (*ssh.ClientConfig, error) {
 	return clientConfig, nil
 }
 
-func Run(ctx context.Context, client *ssh.Client, command string, stdin io.Reader, stdout io.Writer, stderr io.Writer) error {
+func Run(ctx context.Context, client *ssh.Client, command string, stdin io.Reader, stdout io.Writer, stderr io.Writer, envVars *map[string]string) error {
```
Comment (pascalbreuninger) [inline pkg/ssh/helper.go:93]: why does this have to be a pointer? The default value of `map[string]string` is already `nil`

File: cmd/ssh.go (hunk 1 above)
Comment (aacebedo) [reply cmd/ssh.go]: Corrected

File: pkg/ssh/helper.go (hunk 2 above)
Comment (aacebedo) [reply pkg/ssh/helper.go]: corrected

**Comment by aacebedo** [comment]
I fixed the comment. Will it be possible te review please?

**Comment by aacebedo** [comment]
rebased on main and test E2E tests. It is now working

**Comment by pascalbreuninger** [comment]
@aacebedo merged!

**Pull request merged**